File: python/usage_example.py
import mHT 
from mHT.CPF import cpf_fast, cpf_accurate
from matplotlib import pyplot as plt

x=1; #dimensionless
y=1; #dimensionless
print("the output of the cpf_accurate function")
print(cpf_accurate(x,y))

print("the output of the cpf_fast function")
print(cpf_fast(x,y))

#example parameters of the S(1) 3-0 line od H2 perturbed by Ar (reference 10.1063/5.0139229)
nu0 = 112265.5949;#cm-1
GamD = 35.1e-3;#cm-1
Gam0_Ar = 11.3e-3;#cm-1
Gam2_Ar = 0.374e-3;#cm-1
Shift0_Ar = -26.4e-3;#cm-1
Shift2_Ar = 17.8e-3;#cm-1
NuOptRe_Ar = 72.1e-3;#cm-1
NuOptIm_Ar = -16.1e-3#cm-1
nu=nu0+1;#cm-1

print("the output of the mHT function")
print(mHT.profile(nu0,GamD,Gam0_Ar,Gam2_Ar,Shift0_Ar,Shift2_Ar,NuOptRe_Ar,NuOptIm_Ar,nu))

#optional parameters
Sw = 1.0; #dimensionless
Ylm = 1.0e-3; #dimensionless
Xlm = 0.5e-3; #dimensionless
alpha_Ar = 20; #dimensionless

print("the output of the mHT function with optional parameters")
print(mHT.profile(nu0,GamD,Gam0_Ar,Gam2_Ar,Shift0_Ar,Shift2_Ar,NuOptRe_Ar,NuOptIm_Ar,nu,Sw,Ylm,Xlm,alpha_Ar))

#example parameters of the S(1) 3-0 line od H2 perturbed by He (reference 10.1103/PhysRevA.101.052705)
Gam0_He = 11.7e-3;#cm-1
Gam2_He = 5.4e-3;#cm-1
Shift0_He = 30.5e-3;#cm-1
Shift2_He = 12.4e-3;#cm-1
NuOptRe_He = 38.0e-3;#cm-1
NuOptIm_He = -17.5e-3#cm-1

print("the output of the mHT function")
print(mHT.profile(nu0,GamD,Gam0_He,Gam2_He,Shift0_He,Shift2_He,NuOptRe_He,NuOptIm_He,nu))

#optional parameters
alpha_He = 2; #dimensionless

print("the output of the mHT function with optional parameters")
print(mHT.profile(nu0,GamD,Gam0_He,Gam2_He,Shift0_He,Shift2_He,NuOptRe_He,NuOptIm_He,nu,Sw,Ylm,Xlm,alpha_He))

# Preparing tables for plotting
x    = nu0 + np_linspace(-5*GamD,+5*GamD,1001)
y_Ar = [mHT.profile(nu0,GamD,Gam0_Ar,Gam2_Ar,Shift0_Ar,Shift2_Ar,NuOptRe_Ar,NuOptIm_Ar,nu,Sw,Ylm,Xlm,alpha_Ar) for nu in x]
y_He = [mHT.profile(nu0,GamD,Gam0_He,Gam2_He,Shift0_He,Shift2_He,NuOptRe_He,NuOptIm_He,nu,Sw,Ylm,Xlm,alpha_He) for nu in x]

# Plotting complex output
plt.plot(x,list(map(list,zip(*y_Ar)))[0])
plt.plot(x,list(map(list,zip(*y_Ar)))[1])
plt.grid(True)
plt.title("Real (blue) and imaginary (orange) part for mHT output for H$_2$-Ar")
plt.xlabel("Frequency [cm^(-1)]")
plt.ylabel("mHT")
plt.show()

plt.clf()
# Comparing real parts for H2-Ar and H2-He cases.
plt.plot(x,list(map(list,zip(*y_Ar)))[0])
plt.plot(x,list(map(list,zip(*y_He)))[0])
plt.grid(True)
plt.title("Real part of mHT output for H$_2$-Ar (blue) and H$_2$-He (orange).")
plt.xlabel("Frequency [cm^(-1)]")
plt.ylabel("mHT")
plt.show()

File: matlab/usage_example.m
% Importing global variables used in subfunctions
global e pi rp sln2 num0 numinf
load('mhtconstants.mat')

% Example parameters for the CPF functions.
x = 1; % Dimensionless
y = 1; % Dimensionless

% Specifing format output to double precision
format long;

% Displaying the cpf function outputs
disp('The output of the cpf_accurate function:');
disp(cpf_accurate(x,y));
disp('The output of the cpf_fast function:');
disp(cpf_fast(x,y));

% Example parameters of the S(1) 3-0 line of H2 perturbed by Ar (reference: 10.1063/5.0139229) 
nu0        = 112265.5949; % Unperturbed line position in cm-1
GamD       =     35.1e-3; % Doppler broadening in cm-1
Gam0_Ar    =     11.3e-3; % Speed-averaged line-width in cm-1
Gam2_Ar    =     37.4e-5; % Speed dependence of the line-width in cm-1
Shift0_Ar  =    -26.4e-3; % Speed-averaged line-shift in cm-1
Shift2_Ar  =     17.8e-3; % Speed dependence of the line-shift in cm-1
NuOptRe_Ar =     72.1e-3; % Real part of the Dicke parameter in cm-1
NuOptIm_Ar =    -16.1e-3; % Imaginary part of the Dicke parameter in cm-1

% Example frequency point:
nu = nu0 + 1; % Current wavenumber of the computation in cm-1

% Displaying mHT function output
[outRe, outIm] = profile(nu0, GamD, Gam0_Ar, Gam2_Ar, Shift0_Ar, Shift2_Ar, NuOptRe_Ar, NuOptIm_Ar, nu);
disp('The output of the mHT function:');
disp([outRe, outIm]);

% Optional parameters
Ylm      = 1.0e-3; % Dimensionless 
Xlm      = 0.5e-3; % Dimensionless
alpha_Ar = 20;     % Dimensionless

% Displaying mHT function output with opt parameters
[outRe, outIm] = profile(nu0, GamD, Gam0_Ar, Gam2_Ar, Shift0_Ar, Shift2_Ar, NuOptRe_Ar, NuOptIm_Ar, nu, Ylm, Xlm, alpha_Ar);
disp('The output of the mHT function with optional parameters:');
disp([outRe, outIm]);

% Example parameters of the S(1) 3-0 line of H2 perturbed by He (reference 10.1103/PhysRevA.101.052705)
Gam0_He    =  11.7e-3; % cm-1
Gam2_He    =   5.4e-3; % cm-1
Shift0_He  =  30.5e-3; % cm-1
Shift2_He  =  12.4e-3; % cm-1
NuOptRe_He =  38.0e-3; % cm-1
NuOptIm_He = -17.5e-3; % cm-1

% Displaying mHT function output
[outRe, outIm] = profile(nu0, GamD, Gam0_He, Gam2_He, Shift0_He, Shift2_He, NuOptRe_He, NuOptIm_He, nu);
disp('The real-part output of the mHT function (second example):');
disp([outRe, outIm]);

% Optional parameters
alpha_He = 2; % Dimensionless

% Displaying mHT function output with opt parameters
[outRe, outIm] = profile(nu0, GamD, Gam0_He, Gam2_He, Shift0_He, Shift2_He, NuOptRe_He, NuOptIm_He, nu, Ylm, Xlm, alpha_He);
disp('The output of the mHT function with optional parameters (second example):');
disp([outRe, outIm]);

% Preparing tables for plotting
x  = -5*GamD + nu0:GamD/100:5*GamD + nu0;
yR_Ar = zeros(1,length(x));
yI_Ar = zeros(1,length(x));
yR_He = zeros(1,length(x));
yI_He = zeros(1,length(x));
for i=1:length(x)
    [yR_Ar(i), yI_Ar(i)] = profile(nu0, GamD, Gam0_Ar, Gam2_Ar, Shift0_Ar, Shift2_Ar, NuOptRe_Ar, NuOptIm_Ar, x(i));
    [yR_He(i), yI_He(i)] = profile(nu0, GamD, Gam0_He, Gam2_He, Shift0_He, Shift2_He, NuOptRe_He, NuOptIm_He, x(i));
end

% Plotting primary example
fig1 = figure(1);
fig1.Position = [100 100 640 340];
plot(x,yR_Ar)
hold on
plot(x,yI_Ar)
hold off
title("Primary example","Real (blue) and imaginary (orange) part of the mHT output")
xlabel("Frequency [cm^-^1]")
ylabel("mHT")
grid on

% Plotting secondary example
fig2 = figure(2);
fig2.Position = [750 100 640 340];
plot(x,yR_Ar)
hold on
plot(x,yR_He)
hold off
title("Secondary example","Real part of the H_2-Ar (blue) and the H_2-He (orange) mHT output")
xlabel("Frequency [cm^-^1]")
ylabel("mHT")
grid on
